packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/edge/CustomEdge.tsx

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { memo } from 'react';
+import { BaseEdge, EdgeLabelRenderer, EdgeProps, getSmoothStepPath } from 'reactflow';
+import { GQLLabelStyle } from '../../graphql/subscription/labelFragment.types';
+import { CustomEdgeData } from './CustomEdge.types';
+import { getFontSize, getFontWeight, getTextDecoration } from './EdgeLabel';
+
+export const CustomEdge = memo(
+  ({
+    id,
+    sourceX,
+    sourceY,
+    targetX,
+    targetY,
+    sourcePosition,
+    targetPosition,
+    data,
+    style,
+    markerEnd,
+  }: EdgeProps<CustomEdgeData>) => {
+    const [edgePath] = getSmoothStepPath({
+      sourceX,
+      sourceY,
+      sourcePosition,
+      targetX,
+      targetY,
+      targetPosition,
+    });
+
+    const { beginLabel, endLabel, centerLabel } = data;
+
+    // this is a little helper component to render the actual edge label
+    function EdgeLabel({
+      transform,
+      label,
+      edgeLabelStyle,
+    }: {
+      transform: string;
+      label: string;
+      edgeLabelStyle: GQLLabelStyle;
+    }) {
+      return (
+        <div
+          style={{
+            position: 'absolute',
+            background: 'transparent',
+            padding: 10,
+            fontSize: getFontSize(edgeLabelStyle),
+            fontWeight: getFontWeight(edgeLabelStyle),
+            textDecoration: getTextDecoration(edgeLabelStyle),
+            transform,
+            color: edgeLabelStyle.color,
+            zIndex: 999,
```

Comment:
Is this really needed?

Reply:
Are you talking about zIndex ? I needed this to see the labels in front of the nodes. There is a mention of zIndex here https://github.com/eclipse-sirius/sirius-components/issues/2079, I think this can be figured out later.

Reply:
👍